src/shared/lib/MutationUtils.ts

```diff
@@ -0,0 +1,93 @@
+import {
+    default as getCanonicalMutationType, CanonicalMutationType,
+    ProteinImpactType, getProteinImpactTypeFromCanonical
+} from "./getCanonicalMutationType";
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import {findFirstMostCommonElt} from "./findFirstMostCommonElt";
+
+export interface IProteinImpactTypeColors
+{
+    missenseColor: string;
+    inframeColor: string;
+    truncatingColor: string;
+    otherColor: string;
+}
+
+export const DEFAULT_PROTEIN_IMPACT_TYPE_COLORS: IProteinImpactTypeColors = {
+    missenseColor: "#008000",
+    inframeColor: "#8B4513",
+    truncatingColor: "#000000",
+    otherColor: "#8B00C9"
+};
+
+export const MUTATION_TYPE_PRIORITY: {[canonicalMutationType: string]: number} = {
+    "missense": 1,
+    "inframe": 2,
+    "truncating": 4,
+    "nonsense": 6,
+    "nonstop": 7,
+    "nonstart": 8,
+    "frameshift": 4,
+    "frame_shift_del": 4,
+    "frame_shift_ins": 5,
+    "in_frame_ins": 3,
+    "in_frame_del": 2,
+    "splice_site": 9,
+    "fusion": 10,
+    "silent": 11,
+    "other": 11
+};
+
+export function mutationTypeSort(typeA: CanonicalMutationType, typeB: CanonicalMutationType)
+{
+    const priorityA = MUTATION_TYPE_PRIORITY[typeA];
+    const priorityB = MUTATION_TYPE_PRIORITY[typeB];
+    if (priorityA < priorityB) {
+        return -1;
+    } else if (priorityA > priorityB) {
+        return 1;
+    } else {
+        return typeA.localeCompare(typeB);
+    }
+}
+
+export function getColorBasedOnProteinImpactType<T extends IProteinImpactTypeColors>(mutations: Mutation[],
```

Comment:
Actually, since I already explicitly added `extends IProteinImpactTypeColors` to the related component props, I think we don't need a generic type at this point.